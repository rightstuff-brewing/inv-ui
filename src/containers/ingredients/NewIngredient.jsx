
import React from 'react';
import {Button, ControlLabel, FormControl, FormGroup, HelpBlock, Modal} from 'react-bootstrap';

class NewIngredient extends React.Component {
    constructor(props) {
        super(props);

        this.state = {value: ''};
        this.getValidationState = this.getValidationState.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    getValidationState(controlId) {
        if (controlId === 'ingredientName') {
            return 'success';
        }
        else if (controlId === 'ingredientType') {
            return 'success';
        }
        else if (controlId === 'ingredientUrl') {
            return 'success';
        }
    }

    handleChange(e) {
        this.setState({ value: e.target.value });
    }
    
    render() {
        return (
            <Modal show={this.props.show} onHide={this.props.close}>
                <Modal.Header closeButton>
                    <Modal.Title>New Ingredient</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <form>
                        <FormGroup
                            controlId="ingredientName"
                            validationState={this.getValidationState('ingredientName')}
                        >
                            <ControlLabel>Ingredient Name</ControlLabel>
                            <FormControl
                                type="text"
                                value={this.state.value}
                                placeholder="Enter the name of the ingredient"
                                onChange={this.handleChange}
                            />
                            <FormControl.Feedback />
                            <HelpBlock>The name of the ingredient.</HelpBlock>
                        </FormGroup>
                        <FormGroup
                            controlId="ingredientType"
                            validationState={this.getValidationState('ingredientType')}
                        >
                            <ControlLabel>Ingredient Type</ControlLabel>
                            <FormControl componentClass="select" placeholder="select">
                                <option value="malt">Malt</option>
                                <option value="hop">Hops</option>
                                <option value="yeast">Yeast</option>
                            </FormControl>
                        </FormGroup>
                        <FormGroup
                            controlId="ingredientUrl"
                            validationState={this.getValidationState('ingredientUrl')}
                        >
                            <ControlLabel>Ingredient Source Url</ControlLabel>
                            <FormControl 
                                type="text"
                                placeholder="Enter the manufacturer URL for the Ingredient."
                            />
                        </FormGroup>
                        <Button type="submit">
                            Submit
                        </Button>
                    </form>
                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={this.props.close}>Close</Button>
                </Modal.Footer>
            </Modal>
        );
    }
}

export default NewIngredient;