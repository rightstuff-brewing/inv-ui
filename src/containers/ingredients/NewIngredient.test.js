import React from 'react';
import renderer from 'react-test-renderer';
import NewIngredient from './NewIngredient';

// Resolves issue: https://github.com/react-bootstrap/react-bootstrap/issues/2600
jest.mock('react-bootstrap');

describe('NewIngredient component', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(
      <NewIngredient show={false} close={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});