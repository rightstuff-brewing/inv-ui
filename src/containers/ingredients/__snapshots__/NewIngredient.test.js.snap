// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIngredient component matches the snapshot 1`] = `null`;
